feat: added configurable probes for all Astarte components

lucamarchiori opened the pull request:
resolves https://github.com/astarte-platform/astarte-kubernetes-operator/issues/387

This change introduces the ability for users to define custom liveness, readiness, and startup probes for each Astarte component directly within the `Astarte` custom resource.
- The internal probe handling logic has been refactored into a dedicated `probes.go` file to centralize and simplify probe configuration.
- Default probes are still provided if no custom ones are set.
- Tests and changelog are updated accordingly.

lucamarchiori committed:
feat: added configurable probes for all Astarte components

resolves https://github.com/astarte-platform/astarte-kubernetes-operator/issues/387

This change introduces the ability for users to define custom liveness, readiness, and startup probes for each Astarte component directly within the `Astarte` custom resource.
- The internal probe handling logic has been refactored into a dedicated `probes.go` file to centralize and simplify probe configuration.
- Default probes are still provided if no custom ones are set.
- Tests and changelog are updated accordingly. 


Signed-off-by: Luca Marchiori <luca.marchiori@secomind.com>

matt-mazzucato reviewed (commented):
Tiny doubt regarding function naming conventions.

Also, since we're giving users the chance to override Probes, what about improving the documentation of the *Probe fields in the CR (i.e. maybe providing details of the default probes)?

lucamarchiori commented on internal/reconcile/astarte_generic_backend.go:
```diff
@@ -145,8 +138,9 @@ func getAstarteGenericBackendPodSpec(deploymentName string, replicaIndex, replic
-				LivenessProbe:   getAstarteBackendProbe(component, customProbe),
+				ReadinessProbe:  getAstarteReadinessProbe(component, backend),
+				LivenessProbe:   getAstarteLivenessProbe(component, backend),
+				StartupProbe:    getAstarteStartupProbe(backend),
```
LGTM! I'll proceed with the renaming and documentation update

lucamarchiori committed:
refactor: rename probes functions of Astarte components

Signed-off-by: Luca Marchiori <luca.marchiori@secomind.com>

lucamarchiori committed:
feat: add configurable probes for VerneMQ and CFSSL
Implementation note: getters have been separated by component and probe
kind to allow defult value customization in the future.

Signed-off-by: Luca Marchiori <luca.marchiori@secomind.com>

lucamarchiori committed:
chore: document probes in Astarte CR

Signed-off-by: Luca Marchiori <luca.marchiori@secomind.com>

lucamarchiori committed:
feat: add unit/integration tests for VerneMQ and CFSSL probes

Signed-off-by: Luca Marchiori <luca.marchiori@secomind.com>
